database/sql/src/main/scala/cromwell/database/slick/SlickDatabase.scala

```diff
@@ -175,7 +177,7 @@ abstract class SlickDatabase(override val originalDatabaseConfig: Config) extend
     //database.run(action) <-- See comment above private val actionThreadPool
     Future {
       try {
-        Await.result(database.run(action), Duration.Inf)
+        Await.result(database.run(action), queryTimeout)
```

Comment:
I'd prefer to set one timeout so they're not racing.   Whoever reverts this would have to wait somewhere else and ideally doesn't throw this logic away.

Reply:
👍 

ToL: I'm pretty sure this code will hopefully make cromwell more responsive, but will not yield the underlying connection back to the pool, AFAIK, since [Future is not cancellable](https://viktorklang.com/blog/Futures-in-Scala-protips-6.html). So while cromwell will know to move on, the connection may still be stuck underneath.

As this is just a wrapper that I hope to see go away, I'm not too worried about adding this timeout handling in this particular spot. We have no evidence yet of a dangling connection problem, so if we discover that slick/hikari/mysql-client/jdbc/tcp etc. are not closing the connection after some timeout, we can revisit calling `setQueryTimeout` explicitly to clean up dangling connections.

Reply:
Oh I see, that's a valid point.  Your suggestion is the better way to go.  Maybe I'll make the Await something like double the timeout so we don't have a race